organizations/backends/forms.py

```diff
@@ -47,6 +47,13 @@ def __init__(self, *args, **kwargs):
         super(UserRegistrationForm, self).__init__(*args, **kwargs)
         self.initial['username'] = ''
 
+    def clean(self):
+        password = self.cleaned_data.get("password")
+        password_confirm = self.cleaned_data.get("password_confirm")
+        if password != password_confirm or not password:
+            raise forms.ValidationError("Your password entries must match")
```

Comment:
Should use `ugettext` to translate user strings as in the [main forms module](https://github.com/bennylope/django-organizations/blob/dev/organizations/forms.py#L60).